src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -971,50 +971,58 @@ LRESULT CALLBACK FancyZones::s_WndProc(HWND window, UINT message, WPARAM wparam,
 
 void FancyZones::UpdateZoneWindows() noexcept
 {
+    // Mapping between display device name and device index (operating system identifies each display device in the current session with an index value).
+    std::unordered_map<std::wstring, DWORD> displayDeviceIdx;
+    struct capture
+    {
+        FancyZones* fancyZones;
+        std::unordered_map<std::wstring, DWORD>* displayDeviceIdx;
+    };
+
     auto callback = [](HMONITOR monitor, HDC, RECT*, LPARAM data) -> BOOL {
+        capture* params = reinterpret_cast<capture*>(data);
         MONITORINFOEX mi;
         mi.cbSize = sizeof(mi);
-        if (GetMonitorInfo(monitor, &mi))
+        if (GetMonitorInfoW(monitor, &mi))
         {
-            DISPLAY_DEVICE displayDevice = { sizeof(displayDevice) };
-            PCWSTR deviceId = nullptr;
+            auto& displayDeviceIdx = *(params->displayDeviceIdx);
+            FancyZones* fancyZones = params->fancyZones;
 
-            bool validMonitor = true;
-            if (EnumDisplayDevices(mi.szDevice, 0, &displayDevice, 1))
+            DISPLAY_DEVICE displayDevice{ .cb = sizeof(displayDevice) };
+            std::wstring deviceId{};

```

Comment:
`std::wstring deviceId;` works the same here, since `std::wstring`, unlike winAPI structs like `DISPLAY_DEVICE`, isn't a [POD](https://en.cppreference.com/w/cpp/named_req/PODType), but a class with a default constructor that will be executed regardless whether the curly brackets there or not.

Reply:
done